tests/e2e/vsan_stretched_cluster.go

```diff
@@ -363,4 +378,503 @@ var _ = ginkgo.Describe("[vsan-stretch-vanilla] vsan stretched cluster tests", f
+								fmt.Sprintf(
+									"PodVM with vmUUID: %s still exists. So volume: %s is not detached from the PodVM",
+									vmUUID, ss2pod.Spec.NodeName))
+						}
+					}
+				}
+			}
+		}
+
+		// After scale down of sts2, verify the attached volumes match those in CNS Cache
+		for _, ss2pod := range ss2PodsAfterScaleDown.Items {
+			_, err := client.CoreV1().Pods(namespace).Get(ctx, ss2pod.Name, metav1.GetOptions{})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			for _, volumespec := range ss2pod.Spec.Volumes {
+				if volumespec.PersistentVolumeClaim != nil {
+					pv := getPvFromClaim(client, statefulset2.Namespace, volumespec.PersistentVolumeClaim.ClaimName)
+					err := verifyVolumeMetadataInCNS(&e2eVSphere, pv.Spec.CSI.VolumeHandle,
+						volumespec.PersistentVolumeClaim.ClaimName, pv.ObjectMeta.Name, ss2pod.Name)
+					gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				}
+			}
+		}
+
+		// Scaling up statefulset sts1
+		replicas1 += 2
+		ginkgo.By(fmt.Sprintf("Scaling up statefulset: %v to number of Replica: %v", statefulset1.Name, replicas1))
+		_, scaleupErr := fss.Scale(client, statefulset1, replicas1)
+		gomega.Expect(scaleupErr).NotTo(gomega.HaveOccurred())
+		fss.WaitForStatusReplicas(client, statefulset1, replicas1)
+		fss.WaitForStatusReadyReplicas(client, statefulset1, replicas1)
+
+		ss1PodsAfterScaleUp = fss.GetPodList(client, statefulset1)
+		gomega.Expect(ss1PodsAfterScaleUp.Items).NotTo(gomega.BeEmpty(),
+			fmt.Sprintf("Unable to get list of Pods from the Statefulset: %v", statefulset1.Name))
+		gomega.Expect(len(ss1PodsAfterScaleUp.Items) == int(replicas1)).To(gomega.BeTrue(),
+			"Number of Pods in the statefulset: %v should match with number of replicas: %v", statefulset1.Name, replicas1)
+
+		// Scaling down statefulset sts2
+		replicas2 -= 2
+		ginkgo.By(fmt.Sprintf("Scaling down statefulset: %v to number of Replica: %v", statefulset2.Name, replicas2))
+		_, scaledownErr := fss.Scale(client, statefulset2, replicas2)
+		gomega.Expect(scaledownErr).NotTo(gomega.HaveOccurred())
+		fss.WaitForStatusReplicas(client, statefulset2, replicas2)
+		fss.WaitForStatusReadyReplicas(client, statefulset2, replicas2)
+
+		ss2PodsAfterScaleDown = fss.GetPodList(client, statefulset2)
+		gomega.Expect(ss2PodsAfterScaleDown.Items).NotTo(gomega.BeEmpty(),
+			fmt.Sprintf("Unable to get list of Pods from the Statefulset: %v", statefulset2.Name))
+		gomega.Expect(len(ss2PodsAfterScaleDown.Items) == int(replicas2)).To(gomega.BeTrue(),
+			"Number of Pods in the statefulset: %v should match with number of replicas: %v", statefulset2.Name, replicas2)
+
+		scaleDownNDeleteStsDeploymentsInNamespace(ctx, client, namespace)
+
+		ginkgo.By("Bring up the primary site")
+		siteRestore(true)
+
+		ginkgo.By("Wait for k8s cluster to be healthy")
+		// wait for the VMs to move back
+		err = waitForAllNodes2BeReady(ctx, client, pollTimeout*4)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
```

Comment:
Volumes needs to be deleted explicitly

Reply:
While resolving the issue - do brief on the fix that you are making 
Else - it makes the reviewers to review the whole code fully again

Reply:
Sure, will do it from next time. please check line 664-680